Use protobuf on k8s client

aanm opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7414)
<!-- Reviewable:end -->

aanm committed:
k8s: add method to create default Cilium K8s Client

Signed-off-by: André Martins <andre@cilium.dev>

aanm committed:
k8s: add protobuf by default for k8s client

As k8s types support protobuf we can make use of it but we still need to
leave the k8s client for cilium types unmodified since there is no
support on Custom Resources for protobuf.

Signed-off-by: André Martins <andre@cilium.dev>

nebril commented on pkg/k8s/annotate.go:
```diff
@@ -35,6 +36,11 @@ type K8sClient struct {
 	kubernetes.Interface
 }
 
+// K8sCiliumClient is a wrapper around cilium.Interface.
```
not sure, but `s/cilium.Interface/clientset.Interface` ?

nebril reviewed (changes requested):
One typo, but feel free to disregard review if I am wrong on that account

aanm committed:
k8s: add benchmarks tests between protobuf and json

```
$ INTEGRATION=1 PARALLEL_CLIENTS=8 go test --check.vv --check.b --check.bmem --check.f="Benchmark_GetNodes*"
PASS: cnp_test.go:705: K8sIntegrationSuite.Benchmark_GetNodesJSON            500          21234521 ns/op           41773 B/op        471 allocs/op

PASS: cnp_test.go:687: K8sIntegrationSuite.Benchmark_GetNodesProto           500          21231084 ns/op           31172 B/op        303 allocs/op
```

doing a single GET results in the following:

```
JSON     - Request 241 + Response 3816 = 4057 Bytes
Protobuf - Request 260 + Response 2744 = 3004 Bytes

total on wire with protobuf: -26%
only response on wire with protobuf: -39%
```

Signed-off-by: André Martins <andre@cilium.dev>
